src/core/crypto.c

```diff
@@ -893,14 +898,45 @@ int32_t Crypto_Check_Anti_Replay(SecurityAssociation_t* sa_ptr, uint8_t* arsn, u
         // Valid IV received, increment stored value
         else
         {
-            IV_VALID = CRYPTO_TRUE;
+            *IV_VALID = CRYPTO_TRUE;
             // memcpy(sa_ptr->iv, iv, sa_ptr->iv_len);
         }
     }
-    // IV length is greater than zero, but not using an incrementing IV as in GCM
-    // we can't verify this internally as Crpytolib doesn't track previous IVs
-    // or generate random ones
-    // else{}
+    return status;
+}
+
+/*
+** @brief: Check IVs and ARSNs to ensure within valid positive window if applicable
+*/
+int32_t Crypto_Check_Anti_Replay(SecurityAssociation_t* sa_ptr, uint8_t* arsn, uint8_t* iv)
+{
+    int32_t status = CRYPTO_LIB_SUCCESS;
+    int8_t IV_VALID = -1;
+    int8_t ARSN_VALID = -1;
+
+    // Check for NULL pointers
+    status = Crypto_Check_Anti_Replay_Verify_Pointers(sa_ptr, arsn, iv);
+    if(status != CRYPTO_LIB_SUCCESS)
+    {
+        mc_if->mc_log(status);
+        return status;   
```

Comment:
Should we reformat this section to have one extra `if(status !=` at the end but have each `status = Crypto_*` inside these checks to avoid multiple returns?

Reply:
I like this idea. As a note we have 'multiple returns' as a code item. I think we flip-flopped a few times on single return vs return at the closest 'convenient' location.

Reply:
We can do something like this, but that will re-increase the complexity.  I wasn't focusing fully on the single return issue during this as it opened too many other rabbit holes.  I focused solely on complexity.  Do we want to fix issues like that in this issue as well?

Reply:
I misunderstood what you were asking here.  I'll fix this.

Reply:
Fixed -- In upcoming push.